### ISSUE TITLE
prefer-destructuring reports false positive on object bracket notation

### ISSUE THREAD

**Issue by tinovyatkin** [eslint/eslint#7883]
**Tell us about your environment**

* **ESLint Version:**
3.13.0
* **Node Version:**
7.4
* **npm Version:**
4.1.1
**What parser (default, Babel-ESLint, etc.) are you using?**
deafult
**Please show your full configuration:**

```
"env": {
    "es6": true
  },
  "parserOptions": {
    "ecmaVersion": 2017,
    "sourceType": "module",
    "ecmaFeatures": {
      "modules": true
    }
  },
  "plugins": [
    "prefer-spread",
    "node"
  ],
  "rules": {
    "prefer-destructuring": [
      "error",
      {
        "array": true,
        "object": true
      },
      {
        "enforceForRenamedProperties": true
      }
    ]
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
Object.keys(data).map((airlineName) => {
    const { flights, code: airlineCode, airlineLogoUrl, terminal } = data[airlineName];

})
```

**What did you expect to happen?**
No error.
**What actually happened? Please include the actual, raw output from ESLint.**
``Use object destructuring (prefer-destructuring)'
at: '19,11'
``

**Comment by not-an-aardvark** [comment]
I'm not sure, but I think this is working as intended. You can replace the line with

```js
const { [airlineName]: { flights, code: airlineCode, airlineLogoUrl, terminal } } = data;
```

**Comment by evilebottnawi** [comment]
@not-an-aardvark what about `index` access?
```js
const {
    deprecations,
    invalidOptionWarnings,
    warnings
} = results[0];
```

**Comment by not-an-aardvark** [comment]
```js
const [{
    deprecations,
    invalidOptionWarnings,
    warnings
}] = results;
```

**Comment by vitorbal** [comment]
Closing this issue as it looks like the question has been answered. Please feel free to visit us in the [ESLint Gitter](https://gitter.im/eslint/eslint) if you have any other issues!
